Issues/6324 fix attribute parsing

aerych opened the pull request:
Fixes #6324 
Its disappointing, but the attribute retrieval using NSScanner isn't going to be robust enough to handle different possible ways attributes can be written in a tag.  This PR switch to regex, using capture groups to retrieve attribute key/value pairs. Less code and less parsing FTW.

To test: 
Follow `flightpathblog.com` and view its stream in the reader.
Confirm the bug by viewing one of the affected photo posts, such as the one titled "Shoes".
Switch to this branch and confirm the fix. 

Needs review: @nheagy could I trouble you with this one?

nheagy commented on WordPress/Classes/ViewRelated/Views/WPRichText/WPRichTextFormatter.swift:
```diff
@@ -354,6 +354,7 @@ class ListTagProcessor: HtmlTagProcessor
 class AttachmentTagProcessor: HtmlTagProcessor
 {
     let textAttachmentIdentifier = "WPTEXTATTACHMENTIDENTIFIER"
+    static let attributeRegex = try! NSRegularExpression(pattern: "([a-zA-Z-]+)=(?:\"|')([^\"']+)(?:\"|')", options: .CaseInsensitive)
```
Is there a reason to use both `[a-zA-Z]` and `.CaseInsensitive`?

aerych replied:
Not really. I included the cap range more out of habit than anything.  Removing :)

nheagy commented on WordPress/WordPressTest/WPRichTextFormatterTests.swift:
```diff
@@ -89,4 +89,18 @@ class WPRichTextFormatterTests: XCTestCase {
+        XCTAssert(attributes.count == 3)
+        XCTAssert(attributes["src"] != nil)
+        XCTAssert(attributes["alt"] != nil)
+        XCTAssert(attributes["class"] != nil)
```
I'd like to see the values tested as well, to ensure they're correct. Thoughts?

aerych commented:
Thanks @nheagy!  The question about the regex pattern was a good one. The extra cap range wasn't really needed so its good to remove it.  I've updated the tests to check for the correct values, added some emoji for good measure, and added a new test to ensure the attribute key is always lower cased.  Ready for another look sir!

nheagy commented:
This looks great! :shipit:

aerych commented:
Thanks @nheagy !!!
